Resources/public/js/pimcore/objects/tags/objectBridge.js

```diff
@@ -97,145 +98,193 @@ pimcore.object.tags.objectBridge = Class.create(pimcore.object.tags.objects, {
     getColumnFromLayout: function (classNameText, layout, readOnly, prefix) {
         var editor = null,
             renderer = null,
-            minWidth = 40;
+            minWidth = 40,
+            filter = {type: null};
 
         readOnly = (readOnly || layout.readOnly);
         prefix = ((prefix + ' ') || (''));
 
-        if (layout.fieldtype === "input" && !readOnly) {
-            editor = {
-                xtype: 'textfield',
-                allowBlank: !layout.mandatory
-            };
-            renderer = this.renderWithValidation;
-        }
-        else if (layout.fieldtype === "numeric" && !readOnly) {
-            renderer = this.renderWithValidation;
-            var decimalPrecision = Ext.isNumeric(this.fieldConfig.decimalPrecision) ? this.fieldConfig.decimalPrecision : 2;
-            editor = {
-                xtype: 'numberfield',
-                decimalPrecision: decimalPrecision,
-                allowBlank: !layout.mandatory
-            };
-        }
-        else if (layout.fieldtype === "checkbox") {
-            // There seems to be a problem with Ext checkbox column
-            // As a workaround we will skip the composition of the element and return
-            // it as soon as possible meaning all general stuff on bottom of
-            // this function will be passed directly to this column
-
-            var checkBoxColumn = Ext.create('Ext.grid.column.Check', {
-                text: layout.title,
-                width: 40,
-                align: 'left',
-                hidden: !!layout.hidden,
-                sortable: true,
-                dataIndex: classNameText + '_' + layout.name
-            });
+        switch (layout.fieldtype) {
```

Comment:
It would be great to have here a dynamic fallback, to prevent that you have to include hardcoded dependency of fieldtypes. If its possible to have a possibility to have a fallback inside the logic to the fieldtype class, that would make this more powerfull and enables you to include client specific fieldtypes too.

Reply:
Hey Joris! I was also thinking of a way to make this more dynamic when I ran into it. Alternatively you could make it configurable perhaps, or create some kind of hook to allow adding custom fields. Once we run into a case where it's needed it can always be added.

Reply:
Yes wanted this already in the past and was thinking about it. I think you have to register it as a global array that links to the correct class and method, for the default pimcore input types you can build a bridge that fill that array.